src/components/datepicker/js/datepickerDirective.js

```diff
@@ -387,9 +379,34 @@
         }
       });
     }
+
+    // For Angular 1.4 and older, where there are no lifecycle hooks but bindings are pre-assigned,
+    // manually call the $onInit hook.
+    if (angular.version.major === 1 && angular.version.minor <= 4) {
```

Comment:
We can improve on that, sure, but let's not delay this PR further as we need it landed for the Core 1.6 sync.